modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java

```diff
@@ -141,11 +135,39 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
                 if (operation.returnType == null) {
                     operation.returnType = "Void";
                 }
+                if (operation.allParams.stream().anyMatch(p -> p.isFormParam && p.isFile)) {
+                    // If there is a file upload, exclude other form params since it's not clear how the user should access to these
+                    operation.allParams = operation
+                        .allParams
+                        .stream()
+                        .filter(p -> !p.isFormParam || p.isFile)
+                        .collect(Collectors.toList());
+                } else if (operation.allParams.stream().anyMatch(p -> p.isFormParam)) {
+                    // In Vert.x 4 Web OpenAPI the forms are handled as single json object
+                    // We create a dummy param here and remove the other ones
+                    CodegenParameter dummyParam = new CodegenParameter();
+                    dummyParam.isFormParam = true;
+                    dummyParam.isFile = false;
+                    dummyParam.dataType = "JsonObject";
+                    dummyParam.paramName = "formBody";
+                    operation.allParams = Stream.concat(
+                        operation.allParams.stream().filter(p -> !p.isFormParam),
+                        Stream.of(dummyParam)
+                    ).collect(Collectors.toList());
+                }
             }
         }
         return newObjs;
     }
 
+    @Override
+    public void postProcessParameter(CodegenParameter parameter) {
+        super.postProcessParameter(parameter);
+        if (parameter.isUuid || parameter.isDate || parameter.isDateTime) {
+            parameter.dataType = "String";
```

Comment:
@slinkydeveloper what about updating `typeMapping` instead to map these types (e.g. UUID) with "String"?

Reply:
WDYM? Vert.x doesn't have any "strong" mapping for uuid/date/datetime

Reply:
I'll submit a separate PR to show you how I would have done it.

Reply:
I've filed https://github.com/OpenAPITools/openapi-generator/pull/7457. Please take a look when you've time.